### PR TITLE
Convert unordered_multiset to use device_uvector

### PR DESCRIPTION
**Pull request by harrism**
Converts `unordered_multiset` to use device_uvector instead of device_vector. Also adds a cudf::contains benchmark to SEARCH_BENCHMARK.

Contributes to #7287

Performance of `cudf::contains`, the only user of this class, is significantly improved:

```
(rapids) rapids@compose:~/cudf/cpp/build/release$ _deps/benchmark-src/tools/compare.py benchmarks ~/cudf/cpp/build/contains_before.json ~/cudf/cpp/build/contains_after.json 
Comparing /home/mharris/rapids/cudf/cpp/build/contains_before.json to /home/mharris/rapids/cudf/cpp/build/contains_after.json
Benchmark                                                             Time             CPU      Time Old      Time New       CPU Old       CPU New
--------------------------------------------------------------------------------------------------------------------------------------------------
Search/ColumnContains_AllValid/1024/manual_time                    -0.2608         -0.2074             0             0             0             0
Search/ColumnContains_AllValid/4096/manual_time                    -0.9039         -0.8854             1             0             1             0
Search/ColumnContains_AllValid/32768/manual_time                   -0.3135         -0.2648             0             0             0             0
Search/ColumnContains_AllValid/262144/manual_time                  -0.7520         -0.7421             0             0             0             0
Search/ColumnContains_AllValid/2097152/manual_time                 -0.2323         -0.2516             4             3             4             3
Search/ColumnContains_AllValid/16777216/manual_time                -0.1821         -0.1856            40            32            40            32
Search/ColumnContains_AllValid/67108864/manual_time                -0.1368         -0.1377            80            69            81            69
Search/ColumnContains_Nulls/1024/manual_time                       -0.2451         -0.1925             0             0             0             0
Search/ColumnContains_Nulls/4096/manual_time                       -0.2166         -0.1702             0             0             0             0
Search/ColumnContains_Nulls/32768/manual_time                      -0.1798         -0.1450             0             0             0             0
Search/ColumnContains_Nulls/262144/manual_time                     -0.1208         -0.1009             0             0             0             0
Search/ColumnContains_Nulls/2097152/manual_time                    -0.2312         -0.2696             3             2             3             2
Search/ColumnContains_Nulls/16777216/manual_time                   -0.2898         -0.2896            25            17            25            17
Search/ColumnContains_Nulls/67108864/manual_time                   -0.0884         -0.0891            68            62            68            62
```

### CODE REVIEW

File: cpp/benchmarks/search/search_benchmark.cu [hunk 1]
```diff
@@ -153,3 +138,53 @@ BENCHMARK_REGISTER_F(Search, Table)
...
+
+  for (auto _ : state) {
+    cuda_event_timer timer(state, true);
+    auto col = cudf::contains(column, values);
+  }
+}
+
+BENCHMARK_DEFINE_F(Search, ColumnContains_AllValid)(::benchmark::State& state)
+{
+  BM_column(state, false);
```
Comment (devavret) [inline cpp/benchmarks/search/search_benchmark.cu:173]: Did you mean `BM_contains`?

**Review by devavret — CHANGES_REQUESTED** [review]
Everything else is fine.

File: cpp/benchmarks/search/search_benchmark.cu (hunk 1 above)
Comment (harrism) [reply cpp/benchmarks/search/search_benchmark.cu:173]: Hah, perhaps I'm not benchmarking what I intended to!

Comment (harrism) [reply cpp/benchmarks/search/search_benchmark.cu]: Fixed. Now that I'm benchmarking the right thing, performance is dramatically improved!

**Review by hyperbolic2346 — APPROVED** [review]
Happy to hear performance improved. Looks good to me.

**Review by ttnghia — APPROVED** [review]

**Review by devavret — APPROVED** [review]

**Comment by harrism** [comment]
@gpucibot merge

**Pull request merged**